libkbfs/crypto_common.go

```diff
@@ -318,50 +317,36 @@ func nextPowerOfTwo(n uint32) uint32 {
 
 const padPrefixSize = 4
 
-// padBlock adds random padding to an encoded block.
+// padBlock adds zero padding to an encoded block.
 func (c CryptoCommon) padBlock(block []byte) ([]byte, error) {
 	blockLen := uint32(len(block))
-	overallLen := nextPowerOfTwo(blockLen)
-	padLen := int64(overallLen - blockLen)
+	totalLen := nextPowerOfTwo(blockLen)
 
-	buf := bytes.NewBuffer(make([]byte, 0, overallLen+padPrefixSize))
+	buf := make([]byte, padPrefixSize+totalLen)
+	binary.LittleEndian.PutUint32(buf, blockLen)
 
-	// first 4 bytes contain the length of the block data
-	if err := binary.Write(buf, binary.LittleEndian, blockLen); err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	// followed by the actual block data
-	buf.Write(block)
-
-	// followed by random data
-	n, err := io.CopyN(buf, rand.Reader, padLen)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-	if n != padLen {
-		return nil, errors.WithStack(
-			kbfscrypto.UnexpectedShortCryptoRandRead{})
-	}
-
-	return buf.Bytes(), nil
+	copy(buf[padPrefixSize:], block)
+	return buf, nil
 }
 
 // depadBlock extracts the actual block data from a padded block.
 func (c CryptoCommon) depadBlock(paddedBlock []byte) ([]byte, error) {
-	buf := bytes.NewBuffer(paddedBlock)
-
-	var blockLen uint32
-	if err := binary.Read(buf, binary.LittleEndian, &blockLen); err != nil {
-		return nil, errors.WithStack(err)
+	paddedLen := len(paddedBlock)
```

Comment:
`totalLen` or `withPaddingLen` might be better than `paddedLen`, since that kind of sounds like the length of the padded section only.